File: Selenium/Activities/Activity3.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

service = FirefoxService(GeckoDriverManager().install())
with webdriver.Firefox(service=service) as driver:
    driver.get("https://v1.training-support.net/selenium/login-form")
    print("Page title is: " + driver.title)

    # Find username and pwd fields
    username = driver.find_element(By.XPATH, "//input[@id='username']")
    password = driver.find_element(By.XPATH, "//input[@id= 'password']")
    #Send credentials
    username.send_keys("admin")
    password.send_keys("password")
    # click on the login button
    driver.find_element(By.XPATH, "//button[@class = 'ui button']").click()
    #Wait and find the message
    message = driver.find_element(By.XPATH, "//div[@id = 'action-confirmation']")
    print("Success Login message: " + message.text)

    #Close the browser
    driver.close()

File: TestNG/Activities/Activity3.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.Assert;
import org.testng.SkipException;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.DataProvider;
import org.testng.annotations.Test;

public class Activity3 {
    WebDriver driver;
    @BeforeClass
    public void setUp(){
        //Download and install driver
        WebDriverManager.firefoxdriver().setup();
        driver = new FirefoxDriver();

        //Open browser
        driver.get("https://v1.training-support.net/selenium/login-form");
    }

    @Test
    public void homepageTitle() {
        String title = driver.getTitle();
        System.out.println("Home page time is" +title);
        //assert page title
        Assert.assertEquals("Login Form", title);
    }


    @Test
    public void loginTestCase() {
        //Find username and password fields
        WebElement username = driver.findElement(By.id("username"));
        WebElement password = driver.findElement(By.id("password"));

        //Enter values
        username.sendKeys("admin");
        password.sendKeys("password");

        //Click Log in
        driver.findElement(By.cssSelector("button[type='submit']")).click();

        //Assert Message
        String loginMessage = driver.findElement(By.id("action-confirmation")).getText();
        Assert.assertEquals(loginMessage, "Welcome Back, admin");
    }

    @AfterClass
    public void tearDown() {
        driver.quit();
    }


}

